runtime/src/main/java/org/corfudb/runtime/view/ObjectsView.java

```diff
@@ -76,12 +70,12 @@ public void TXBegin() {
 
         /* If it is a nested transaction, inherit type of parent */
         if (TransactionalContext.isInTransaction()) {
-            type = TransactionalContext.getCurrentContext().getBuilder().getType();
+            type = TransactionalContext.getCurrentContext().getTransaction().getType();
```

Comment:
Is it that we don't inherit snapshot from this point because in the case of a TX different than snapshot we might have not yet accessed an object, is this assumption right? should we add a comment on why we do not directly inherit snapshot timestamp at this point?